providers/core.udpsocket.js

```diff
@@ -23,7 +38,7 @@ UDP_Firefox.prototype.bind = function(address, port, continuation) {
 
 UDP_Firefox.prototype.getInfo = function(continuation) {
   var returnValue = {
-    localAddress: "localhost",
+    localAddress: this._nsIUDPSocket.localAddr.address,
     localPort: this._nsIUDPSocket.port
```

Comment:
It may be good to pull port from `this._nsIUDPSocket.localAddr.port` for consistency - from logging the tests it looks to have the same value as `this._nsIUDPSocket.port`, so it should work the same, but look more consistent and maybe inoculate us from weird edge cases.